app/assets/javascripts/mirador_widget_admin.js

```diff
@@ -80,10 +88,32 @@
     // TODO: Add some sort of loading animation and clean it up after
     // complete to help reduce confusion with slow loading manifests
     function fetchSelectedItem(block, manifestUrl) {
+      if (validUrl(block, manifestUrl) === false) {
+        return false
+      }
+
       $.get(manifestUrl)
-       .done(function(data) {
-         block.trigger('item-added', { block: block, manifest: data });
-       });
+        .done(function(data) {
+          if(typeof(data) == 'string') {
+            data = JSON.parse(data);
+          }
```

Comment:
Since we're doing this here, I wonder if we can remove it from the `addIiifItemToSection` function (since this `data` is passed there through the `item-added` event.